libeos-updater-util/avahi-service-file.c

```diff
@@ -248,12 +518,699 @@ eos_avahi_service_file_delete (const gchar   *avahi_service_directory,
+  if (version == 0 || version > 1)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
+                   "unknown TXT record version: %u", version);
+      return FALSE;
+    }
+
+  if (out_version)
+    *out_version = version;
+  return TRUE;
+}
+
+static gboolean
+check_v1_options (GVariantDict  *options_dict,
+                  GError       **error)
+{
+  if (!get_and_check_bloom_size (options_dict, NULL, error))
+    return FALSE;
+
+  if (!get_and_check_bloom_k (options_dict, NULL, error))
+    return FALSE;
+
+  if (!get_and_check_bloom_hash_func_id (options_dict, NULL, error))
+    return FALSE;
+
+  if (!get_and_check_avahi_service_port (options_dict, NULL, error))
+    return FALSE;
+
+  if (!get_and_check_txt_records_size_level (options_dict, NULL, error))
+    return FALSE;
+
+  return TRUE;
+}
+
+/**
+ * eos_ostree_avahi_service_file_check_options:
+ * @options: (nullable): vardict #GVariant
+ * @error: return location for a #GError
+ *
+ * Validates the contents of @options. Unknown keys in @options are
+ * ignored. If some key-value pair in @options is not valid in some
+ * way, the function will fill the @error variable and return %FALSE.
+ *
+ * If @options has a floating reference then this function sinks it.
+ *
+ * Note that this function can not check the real validity of the
+ * #EOS_OSTREE_AVAHI_OPTION_TXT_RECORDS_SIZE_LEVEL_Y key - it only
+ * checks if the key has a valid value, but it is not able to check if
+ * generated TXT records do not break the imposed limit. This error
+ * can be reported only by the
+ * eos_ostree_avahi_service_file_generate() function.
+ *
+ * Returns: whether contents of @options are valid
+ */
+gboolean
+eos_ostree_avahi_service_file_check_options (GVariant  *options,
+                                             GError   **error)
+{
+  g_autoptr(GVariant) reffed_options = (options != NULL) ? g_variant_ref_sink (options) : NULL;
+  g_auto(GVariantDict) options_dict = G_VARIANT_DICT_INIT (reffed_options);
+  guint8 version;
+
+  g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+
+  if (!get_and_check_version (&options_dict, &version, error))
+    return FALSE;
+
+  switch (version)
+    {
+    case 1:
+      return check_v1_options (&options_dict, error);
+
+    default:
+      g_assert_not_reached ();
+    }
+}
+
+/**
+ * eos_ostree_avahi_service_file_generate:
+ * @avahi_service_directory: path to the directory containing `.service` files
+ * @refs_to_advertise: an array of refs to advertise
+ * @summary_timestamp: timestamp of the repo summary
+ * @options: (nullable): vardict #GVariant
+ * @cancellable: (nullable): a #GCancellable
+ * @error: return location for a #GError
+ *
+ * Create a `.service` file in @avahi_service_directory for the updater. This
+ * instructs Avahi to advertise a DNS-SD service for the updater, with TXT
+ * records indicating this machine has the refs for @ostree_path available with
+ * a commit at @head_commit_timestamp.
```

Comment:
This documentation seems outdated — `ostree_path` and `head_commit_timestamp` are no longer arguments to this function. We now have `refs_to_advertise`.